### ISSUE TITLE
`browser` in a single-line function displays outside source file

### ISSUE THREAD

**Issue by OfekShilon** [rstudio/rstudio#12416]
### System details
Tested in two configurations:

    RStudio Edition : server
    RStudio Version : 1.4.1103 and desktop 2022.07.2 Build 576
    OS Version      : ubuntu 18.04
    R Version       : 4.0.5

    RStudio Edition : desktop
    RStudio Version : 2022.07.2 Build 576
    OS Version      : ubuntu 20.04
    R Version       : 4.2.2

### Steps to reproduce the problem
 1. Source a file with these contents:
```r
# tmp.R
singleline <- function() { browser() }

multiline <- function() {
  browser()
}
```
 2. At the console run `multiline()` and debugging starts correctly.
 3. At the console run `singleline()` and debugging starts at a temp file, outside the original source with the original context.

Screen capture attached in a comment.
### Describe the problem in detail
Debugging experience shouldn't depend on whether the function was implemented in a single line or not. As far as I can see both functions have equivalent info in the `srcref` attribute (but I might be wrong).

### Describe the behavior you expected
Break inside the right location _in my source file_ when breaking inside a single-line function.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by OfekShilon** [comment]
https://user-images.githubusercontent.com/73080/206691089-ea7ca402-405d-4036-8b6c-db2c4521b0fb.mp4

**Comment by ronblum** [comment]
@OfekShilon Thank you for noting this! This looks like the same issue as #12505, with the added info that single-line functions don't trigger the behavior. Do you agree?

**Comment by OfekShilon** [comment]
@ronblum The symptom (opening a new tab) is the same, but I don't think this is the same issue.
In #12505 their problem is that *after editing but not sourcing* a source file, rstudio opens a new tab when debugging, since it is debugging the old function which is no longer in the file. This actually seems entirely reasonable to me and I'm not sure what they expect rstudio to do in such case (it _can't_ use the file, the function being debugged is no longer there).

This is not what I'm reporting here: there is no editing involved. The same code, in two forms (multi-line, single-line), one is debugged in the source file while the other (single-line) opens a new tab and loses visual context.

**Comment by OfekShilon** [comment]
@ronblum  the 'repro needed' tag means it is not reproduced for you?

**Comment by ronblum** [comment]
Good question! The `repro needed` label means that I (we) need some more information. In this case, just whether or not you think that the two issues are duplicates. I added a description, so that hovering over the label will explain it.

I tried this out on a four combinations of RStudio Desktop and Server 1.0.153, 2022.07.2+576, and 2022.12.0+353 on MacOS 13.2 and Ubuntu 20.04. I get the same behavior, the new window, for both single- and multi-line functions, using the same code and both if the file is saved or not. But regardless, you're clearly seeing a difference which shouldn't be there, so I'm leaving this open (and not marking it as a duplicate).
